compose.yml

```diff
@@ -1,20 +1,74 @@
+# Docker compose file that allows each of the services we require to start
+# only when their dependent service(s) have started. This is achieved through
+# the use of a docker healthcheck for each service.
+
+# First, construct private network for our services to communicate with each other
+networks:
+  network:
+    driver: bridge
+
 services:
-  neo4j-dbms:
-    build: "data/"
+  # neo4j service
+  neo4j-db:
+    image: neo4j:latest
+    env_file: ./data/.env
+    networks:
+      - network
+    # network ports, we can customise exposed ports so that we
+    # don't conflict with other neo4j instances that may be running.
+    # docker compose expects custom ports to exist in a root level .env file.
```

Comment:
is this configurable? Feels like we're repeating information in .env and data/.env

Reply:
Docker compose can only pull-in config from a root `.env` file.
`data/.env` is only being used to pass in environment config into the "neo4j" container.

One option we have is:
1) Put all environment variables into a root `.env` file
2) Delete the other `.env` files
3) Update Docker compose config and explicitly tell it which environment variables to pass into which containers

That is certainly a pattern used elsewhere.